src/viem/saga.ts

```diff
@@ -199,36 +201,33 @@ export function* sendPayment({
  */
 function* getTransferSimulateContract({
   wallet,
-  tokenId,
+  tokenAddress,
+  tokenInfo,
   amount,
   recipientAddress,
   comment,
   feeInfo,
 }: {
   wallet: ViemWallet
+  tokenAddress: string
+  tokenInfo: TokenBalance
```

Comment:
seems like you could use `TokenBalanceWithAddress` here

Reply:
given that there are ongoing use cases for dealing with tokens that have addresses differently (like right here), we may not ever want to deprecate that type

Reply:
Yeah, this is a good call. I agree we may not want to deprecate this type, but I wonder how we would differentiate between cases where its usage actually _is_ deprecated (i.e., somewhere we truly want to transition over to using the new types) vs this use, which is actually valid.